### PR TITLE
Plat 606 kafka packages refactor

### PR DESCRIPTION
**Pull request by nour-borgi**
In this PR, I refactored the swarm script of: 
- message bus kafka
- kafka mapper consumer 
- kafka unbundler consumer 

Fixed a problem with displaying the warning multiple times: 
image

to be displayed only once: 
image

### CODE REVIEW

File: kafka-mapper-consumer/swarm.sh [hunk 1]
```diff
@@ -1,34 +1,73 @@
...
+  try \
+    "docker service scale instant_$service_name=0" \
+    catch \
+    "Failed to scale down $service_name"
+}
+
+function destroy_package() {
+  docker::service_destroy "$service_name"
+
+  docker::prune_configs "kafka-mapper-consumer"
```
Comment (marrouchi) [inline kafka-mapper-consumer/swarm.sh:52]:
```suggestion
  docker::prune_configs "$service_name"
```

File: kafka-mapper-consumer/swarm.sh [hunk 2]
```diff
@@ -1,34 +1,73 @@
...
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/log.sh"
+}
+
+function initialize_package() {
+  (
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml"
+    docker::deploy_sanity "${service_name}"
+  ) || {
+    log error "Failed to deploy Kafka Mapper Consumer package"
```
Comment (marrouchi) [inline kafka-mapper-consumer/swarm.sh:37]:
```suggestion
    log error "Failed to deploy $service_name package"
```

File: kafka-mapper-consumer/swarm.sh [hunk 3]
```diff
@@ -1,34 +1,73 @@
...
 main() {
-  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
-    config::set_config_digests "${COMPOSE_FILE_PATH}"/docker-compose.yml
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml instant" "Failed to deploy Kafka Mapper Consumer"
+  init_vars "$@"
+  import_sources
 
-    config::remove_stale_service_configs "${COMPOSE_FILE_PATH}"/docker-compose.yml "kafka-mapper-consumer"
+  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
+    log info "Running Kafka Mapper Consumer package"
```
Comment (marrouchi) [inline kafka-mapper-consumer/swarm.sh:60]:
```suggestion
    log info "Running $service_name package"
```

File: kafka-mapper-consumer/swarm.sh [hunk 4]
```diff
@@ -1,34 +1,73 @@
...
-    docker::deploy_sanity kafka-mapper-consumer
+    initialize_package
   elif [[ "${ACTION}" == "down" ]]; then
-    try "docker service scale instant_kafka-mapper-consumer=0" "Failed to scale down kafka-mapper-consumer"
+    log info "Scaling down Kafka Mapper Consumer"
+
+    scale_services_down
   elif [[ "${ACTION}" == "destroy" ]]; then
-    docker::service_destroy kafka-mapper-consumer
+    log info "Destroying Kafka Mapper Consumer"
```
Comment (marrouchi) [inline kafka-mapper-consumer/swarm.sh:68]:
```suggestion
    log info "Destroying $service_name"
```

File: kafka-mapper-consumer/swarm.sh [hunk 5]
```diff
@@ -1,34 +1,73 @@
...
 
-    config::remove_stale_service_configs "${COMPOSE_FILE_PATH}"/docker-compose.yml "kafka-mapper-consumer"
+  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
+    log info "Running Kafka Mapper Consumer package"
 
-    docker::deploy_sanity kafka-mapper-consumer
+    initialize_package
   elif [[ "${ACTION}" == "down" ]]; then
-    try "docker service scale instant_kafka-mapper-consumer=0" "Failed to scale down kafka-mapper-consumer"
+    log info "Scaling down Kafka Mapper Consumer"
```
Comment (marrouchi) [inline kafka-mapper-consumer/swarm.sh:64]:
```suggestion
    log info "Scaling down $service_name"
```

**Review by marrouchi — CHANGES_REQUESTED** [review]
I believe the comments should apply across all PRs , wdyt ?

File: kafka-mapper-consumer/swarm.sh (hunk 1 above)
Comment (nour-borgi) [reply kafka-mapper-consumer/swarm.sh:52]: it is config label and not service name

File: kafka-mapper-consumer/swarm.sh (hunk 2 above)
Comment (nour-borgi) [reply kafka-mapper-consumer/swarm.sh:37]: done in another PR

File: kafka-mapper-consumer/swarm.sh (hunk 3 above)
Comment (nour-borgi) [reply kafka-mapper-consumer/swarm.sh:60]: done in another PR

File: kafka-mapper-consumer/swarm.sh (hunk 4 above)
Comment (nour-borgi) [reply kafka-mapper-consumer/swarm.sh:68]: done in another PR

File: kafka-mapper-consumer/swarm.sh (hunk 5 above)
Comment (nour-borgi) [reply kafka-mapper-consumer/swarm.sh:64]: done in another PR

**Review by marrouchi — APPROVED** [review]

**Review by Mark-Labuschagne — APPROVED** [review]

**Pull request merged**